File: LeetCode/Easy/1047_Remove_All_Adjacent_Duplicates_In_String/Solution.py
## 2022-11-09 (stack)
class Solution:
    def removeDuplicates(self, s: str) -> str:
        stack = []
        for c in s:
            if stack and c == stack[-1]:
                stack.pop()
            else:
                stack.append(c)
        return "".join(stack)

File: LeetCode/Easy/1047_Remove_All_Adjacent_Duplicates_In_String/Solution.cpp
// 2022-09-18 (stack)
class Solution {
public:
    string removeDuplicates(string s) {
        deque<char> stack;
        for (char c : s) {
            if (!stack.empty() && stack.back() == c) {
                stack.pop_back();
            } else {
                stack.push_back(c);
            }
        }
        
        string output = "";
        for (auto it = stack.begin(); it != stack.end(); it++)
            output += *it;
        return output;
    }
};